// @acid-info/logos-docusaurus-preset/lib/validateOptions.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.validateOptions = void 0;
const utils_validation_1 = require("@docusaurus/utils-validation");
const types_1 = require("./types");
const schema = utils_validation_1.Joi.object({
    businessUnit: utils_validation_1.Joi.string()
        .valid(...Object.values(types_1.BusinessUnits))
        .default(types_1.BusinessUnits.Logos),
    contactInfo: utils_validation_1.Joi.string().optional(),
    theme: utils_validation_1.Joi.object()
        .schema({
        name: utils_validation_1.Joi.string()
            .valid(...Object.keys(types_1.ThemeNames))
            .required(),
        themeOptions: utils_validation_1.Joi.object().optional().default({}),
    })
        .optional()
        .default({}),
    customSiteConfig: utils_validation_1.Joi.boolean().optional().default(false),
});
const validateOptions = ({ options, validate, }) => validate(schema, options ?? {});
exports.validateOptions = validateOptions;
