// @acid-info/logos-docusaurus-preset/lib/types/status.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.Status = void 0;
var Status;
(function (Status) {
    let Department;
    (function (Department) {
        Department["Blockchain"] = "Blockchain";
        Department["Codex"] = "Codex";
        Department["CommsHub"] = "Comms Hub";
        Department["CommsMarketing"] = "Comms/Marketing";
        Department["DAO"] = "DAO";
        Department["Desktop"] = "Desktop";
        Department["Documentation"] = "Documentation";
        Department["EcoSystemDevelopment"] = "Eco-System-Development";
        Department["Finance"] = "Finance";
        Department["InfrastructureOps"] = "Infrastructure Ops";
        Department["Keycard"] = "Keycard";
        Department["Legal"] = "Legal";
        Department["Libp2p"] = "Libp2p";
        Department["Mobile"] = "Mobile";
        Department["Nim"] = "Nim";
        Department["PPG"] = "PPG";
        Department["PeopleOps"] = "People Ops";
        Department["ProductDesign"] = "Product Design";
        Department["ProgramLead"] = "Program Lead";
        Department["Security"] = "Security";
        Department["WakuProduct"] = "Waku Product";
        Department["WakuResearch"] = "Waku Research";
        Department["Web"] = "Web";
    })(Department = Status.Department || (Status.Department = {}));
})(Status = exports.Status || (exports.Status = {}));
