// @acid-info/logos-docusaurus-preset/lib/theme-config/codex.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.codexThemeConfig = void 0;
const tslib_1 = require("tslib");
const base_1 = tslib_1.__importDefault(require("./base"));
exports.codexThemeConfig = {
    ...base_1.default,
    navbar: {
        ...base_1.default.navbar,
        logo: {
            alt: 'Codex',
            src: 'theme/image/logo-black.svg',
            srcDark: 'theme/image/logo.svg',
            height: 26,
        },
    },
    metadata: [
        { name: 'keywords', content: 'codex, storage' },
        {
            name: 'description',
            content: 'Codex is building a Decentralized Durability Storage',
        },
        { name: 'image', content: 'theme/image/preview-image.png' },
    ],
    footer: {
        ...base_1.default.footer,
        copyright: `Codex © ${new Date().getFullYear()}<br/>All rights reserved.`,
        logo: {
            alt: 'Codex',
            src: 'theme/image/logo.svg',
            href: '/',
            width: 22,
        },
        links: [
            {
                title: 'Community',
                items: [
                    {
                        label: 'Discord',
                        href: '4qdQN5JaWW',
                    },
                    {
                        label: 'Twitter',
                        href: 'codexstorage',
                    },
                ],
            },
        ],
    },
};
exports.default = exports.codexThemeConfig;
