// @acid-info/logos-docusaurus-preset/lib/theme-config/logos.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.logosThemeConfig = void 0;
const tslib_1 = require("tslib");
const base_1 = tslib_1.__importDefault(require("./base"));
exports.logosThemeConfig = {
    ...base_1.default,
    navbar: {
        ...base_1.default.navbar,
        logo: {
            alt: 'Logos',
            src: 'theme/image/logo-black.svg',
            srcDark: 'theme/image/logo.svg',
            height: 26,
        },
    },
    metadata: [
        { name: 'keywords', content: 'Logos' },
        {
            name: 'description',
            content: 'Logos is a grassroots movement to provide trust-minimized, corruption resistant governing services and social institutions to underserved citizens.',
        },
        { name: 'image', content: 'theme/image/preview-image.png' },
    ],
    footer: {
        ...base_1.default.footer,
        copyright: `Logos © ${new Date().getFullYear()}<br/>All rights reserved.`,
        logo: {
            alt: 'Logos',
            src: 'theme/image/logo.svg',
            href: '/',
            width: 22,
        },
        links: [
            {
                title: 'Community',
                items: [],
            },
        ],
    },
};
exports.default = exports.logosThemeConfig;
