// @acid-info/logos-docusaurus-preset/lib/theme-config/nimbus.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.nimbusThemeConfig = void 0;
const tslib_1 = require("tslib");
const base_1 = tslib_1.__importDefault(require("./base"));
exports.nimbusThemeConfig = {
    ...base_1.default,
    navbar: {
        ...base_1.default.navbar,
        logo: {
            alt: 'Nimbus',
            src: 'theme/image/logo-black.svg',
            srcDark: 'theme/image/logo.svg',
            height: 26,
        },
        items: [],
    },
    metadata: [
        { name: 'keywords', content: 'nimbus' },
        {
            name: 'description',
            content: 'Nimbus, a Lighter Ethereum Client',
        },
        { name: 'image', content: 'theme/image/preview-image.png' },
    ],
    footer: {
        ...base_1.default.footer,
        copyright: `Nimbus © ${new Date().getFullYear()}<br/>All rights reserved.`,
        logo: {
            alt: 'Nimbus',
            src: 'theme/image/logo.svg',
            href: '/',
            width: 22,
        },
        links: [
            {
                items: [
                    {
                        label: 'Gitter',
                        href: 'https://gitter.im/status-im/nimbus',
                    },
                    {
                        label: 'Twitter',
                        href: 'https://twitter.com/ethnimbus',
                    },
                    {
                        label: 'Discord',
                        href: 'https://discord.gg/XRxWahP',
                    },
                    {
                        label: 'Status',
                        href: 'https://join.status.im/chat/public/nimbus-general',
                    },
                ],
            },
        ],
    },
};
exports.default = exports.nimbusThemeConfig;
