// @acid-info/logos-docusaurus-preset/lib/theme-config/nomos.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.nomosThemeConfig = void 0;
const tslib_1 = require("tslib");
const base_1 = tslib_1.__importDefault(require("./base"));
exports.nomosThemeConfig = {
    ...base_1.default,
    navbar: {
        ...base_1.default.navbar,
        logo: {
            alt: 'Nomos',
            src: 'theme/image/logo.svg',
            srcDark: 'theme/image/logo.svg',
            height: 26,
        },
        items: [],
    },
    metadata: [
        { name: 'keywords', content: 'Nomos' },
        {
            name: 'description',
            content: "Nomos supports the demands of aspiring network states, defending self-determination at the network level for humanity's underserved.",
        },
        { name: 'image', content: 'theme/image/preview-image.png' },
    ],
    footer: {
        ...base_1.default.footer,
        copyright: `Nomos © ${new Date().getFullYear()}<br/>All rights reserved.`,
        logo: {
            alt: 'Nomos',
            src: 'theme/image/logo.svg',
            href: '/',
            width: 22,
        },
        links: [],
    },
};
exports.default = exports.nomosThemeConfig;
