// @acid-info/logos-docusaurus-preset/lib/theme-config/vac-research.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.vacResearchThemeConfig = void 0;
const tslib_1 = require("tslib");
const base_1 = tslib_1.__importDefault(require("./base"));
exports.vacResearchThemeConfig = {
    ...base_1.default,
    navbar: {
        ...base_1.default.navbar,
        logo: {
            alt: 'Vac Research',
            src: 'theme/image/logo.svg',
            height: 26,
        },
        items: [],
    },
    metadata: [
        {
            name: 'description',
            content: 'Vac - Communication, Privacy, Etc.',
        },
        { name: 'image', content: 'theme/image/preview-image.png' },
    ],
    footer: {
        ...base_1.default.footer,
        copyright: `Vac Research © ${new Date().getFullYear()}<br/>All rights reserved.`,
        logo: {
            alt: 'Vac Research',
            src: 'theme/image/logo.svg',
            href: '/',
            width: 22,
        },
        links: [],
    },
};
exports.default = exports.vacResearchThemeConfig;
