// @acid-info/logos-docusaurus-preset/lib/theme-config/waku.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.wakuThemeConfig = void 0;
const tslib_1 = require("tslib");
const base_1 = tslib_1.__importDefault(require("./base"));
exports.wakuThemeConfig = {
    ...base_1.default,
    navbar: {
        ...base_1.default.navbar,
        logo: {
            alt: 'Waku',
            src: 'theme/image/logo-black.svg',
            srcDark: 'theme/image/logo.svg',
            height: 26,
        },
        items: [],
    },
    metadata: [
        { name: 'keywords', content: 'waku, web3' },
        {
            name: 'description',
            content: 'Waku is the communication layer for Web3. Decentralized communication that scales.',
        },
        { name: 'image', content: 'theme/image/preview-image.png' },
    ],
    footer: {
        ...base_1.default.footer,
        copyright: `Waku © ${new Date().getFullYear()}<br/>All rights reserved.`,
        logo: {
            alt: 'Waku',
            src: 'theme/image/logo.svg',
            href: '/',
            width: 22,
        },
        links: [
            {
                title: 'Community',
                items: [
                    {
                        label: 'Discourse',
                        href: 'https://forum.vac.dev/',
                    },
                    {
                        label: 'Discord',
                        href: 'j5pGbn7MHZ',
                    },
                    {
                        label: 'Twitter',
                        href: 'waku-org',
                    },
                ],
            },
        ],
    },
};
exports.default = exports.wakuThemeConfig;
