// @acid-info/logos-docusaurus-theme/lib/client/lib/themes.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.darkTheme = exports.lightTheme = void 0;
const lsd_react_1 = require("@acid-info/lsd-react");
const typography = {
    h1: {
        fontSize: '2.875rem',
        lineHeight: '3.25rem',
    },
    h2: {
        fontSize: '2rem',
        lineHeight: '2.5rem',
    },
    h3: {
        fontSize: '1.5rem',
        lineHeight: '2rem',
    },
    h4: {
        fontSize: '1.375rem',
        lineHeight: '1.75rem',
    },
};
const typographyMobile = {
    h1: {
        fontSize: '2.25rem',
        lineHeight: '2.75rem',
    },
    h2: {
        fontSize: '1.75rem',
        lineHeight: '2.25rem',
    },
};
const breakpoints = {
    xs: {
        width: 0,
        typography: typographyMobile,
    },
    sm: {
        width: 576,
    },
    md: {
        width: 768,
    },
    lg: {
        width: 997,
        typography,
    },
    xl: {
        width: 1200,
    },
};
exports.lightTheme = (0, lsd_react_1.createTheme)({
    breakpoints,
    palette: {},
    typography,
    typographyGlobal: {},
    name: 'logos-docusaurus-theme-light',
}, lsd_react_1.defaultThemes.light);
exports.darkTheme = (0, lsd_react_1.createTheme)({
    breakpoints: breakpoints,
    palette: {},
    typography,
    typographyGlobal: {},
    name: 'logos-docusaurus-theme-light',
}, lsd_react_1.defaultThemes.dark);
