// @acid-info/logos-docusaurus-theme/lib/client/lib/ui.utils.js
"use strict";
Object.defineProperty(exports, "__esModule", { value: true });
exports.calcHeroInfoMb = exports.isMobile = exports.generateTextShadow = exports.isTouchDevice = exports.calcScrollThreshold = exports.mapFloat = exports.random = void 0;
const random = (min, max) => {
    return Math.random() * (max - min) + min;
};
exports.random = random;
const mapFloat = (value, inMin, inMax, outMin, outMax) => {
    return ((value - inMin) * (outMax - outMin)) / (inMax - inMin) + outMin;
};
exports.mapFloat = mapFloat;
const calcScrollThreshold = () => {
    return window.innerHeight * 0.4;
};
exports.calcScrollThreshold = calcScrollThreshold;
const isTouchDevice = () => {
    if (typeof window === 'undefined')
        return false;
    return ('ontouchstart' in window ||
        navigator.maxTouchPoints > 0 ||
        // @ts-ignore
        (navigator.msMaxTouchPoints && navigator.msMaxTouchPoints > 0));
};
exports.isTouchDevice = isTouchDevice;
const generateTextShadow = (steps) => {
    let shadows = '';
    for (let i = 0; i < steps; i++) {
        let colorValue = Math.floor((i / (steps - 1)) * 255); // Generate a value between 0 and 255
        let shadow = `${-i * 5}px ${-i * 5}px 0px rgb(${colorValue}, ${colorValue}, ${colorValue})`;
        shadows += (i === 0 ? '' : ', ') + shadow;
    }
    return shadows;
};
exports.generateTextShadow = generateTextShadow;
const isMobile = () => {
    if (typeof window === 'undefined')
        return false;
    return window.innerWidth < 997;
};
exports.isMobile = isMobile;
const calcHeroInfoMb = (sy) => {
    return (0, exports.mapFloat)(scrollY, 0, (0, exports.calcScrollThreshold)(), 0, 100);
};
exports.calcHeroInfoMb = calcHeroInfoMb;
